tools/ci/actions/runner.sh

```diff
@@ -18,7 +18,7 @@ set -xe
 
 PREFIX=~/local
 
-MAKE="make $MAKE_ARG"
+MAKE="make --warn-undefined-variables $MAKE_ARG"
```

Comment:
I am not too fond of CI tests that are harsher than the dev version.

Reply:
Except it already runs in CI, but with a @shindere alarm! It's virtually impossible to put this one into the development workflow, because `make` doesn't (I don't believe) give us an easy way to signal an error.

If we'd had this warning in CI, I would have fixed this before opening the PR which broke it (or very shortly after it failed and before it was reviewed)